web_frontend/osmaxx/job_progress/middleware.py

```diff
@@ -29,11 +32,41 @@ def update_order_if_stale(extraction_order):
     get_cached_or_set(
         'extraction_order_{}_progress'.format(extraction_order.id),
         update_order, extraction_order,
-        timeout=timedelta(minutes=1).seconds
+        on_cache_hit=_log_cache_hit,
+        timeout=timedelta(minutes=1).seconds,
     )
 
 
 def update_order(extraction_order):
+    _log_cache_miss(extraction_order)
     conversion_client = get_authenticated_api_client()
     conversion_client.update_order_status(extraction_order)
+    if logger.isEnabledFor(logging.DEBUG):
+        message = "Fetched, updated and cached ExtractionOrder {extraction_order} progress: {progress}".format(
+            extraction_order=extraction_order.id,
+            progress=ExtractionOrderState.label(extraction_order.state),
+        )
+        logger.debug(message)
     return extraction_order.state
+
+
+def _log_cache_miss(extraction_order):
+    if not logger.isEnabledFor(logging.INFO):
+        return
+    message = "Cache miss - Re-fetching progress of pending ExtractionOrder {extraction_order}.".format(
+        extraction_order=extraction_order.id,
+    )
+    logger.info(message)
```

Comment:
I chose `info`, because this is conceptually a mid- to high-level message (something happened or something else) and because we're doing something that might be unusual in the middleware. Thus when we have a reason to view the logs, it's probably something we'd want to be aware of. Someone who only wants to get output for potential problems should set loglevel `warning` and won't see these.